src/rezplugins/shell/cmd.py

```diff
@@ -68,14 +68,11 @@ def get_startup_sequence(cls, rcfile, norc, stdin, command):
         )
 
     @classmethod
+    @syspaths_composer
```

Comment:
So the functionality of syspaths_composer is fine, however I don't think it's correct for it to be a decorator like this.

The intent of `Shell.get_syspaths` is to define what system paths are associated with that shell - not _how_ to then apply them. So I would just take the functionality from the decorator, and apply it only where syspaths are actually set into PATH (https://github.com/nerdvegas/rez/blob/master/src/rez/rex.py#L1124)